google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -244,7 +234,42 @@ public void onCompleted() {
                 "numDocuments", AttributeValue.longAttributeValue(documentReferences.length)));
 
     streamRequest(request.build(), responseObserver, firestoreClient.batchGetDocumentsCallable());
+  }
+
+  /** Internal getAll() method that accepts an optional transaction id. */
+  ApiFuture<List<DocumentSnapshot>> getAll(
+      final DocumentReference[] documentReferences,
+      @Nullable FieldMask fieldMask,
+      @Nullable ByteString transactionId) {
+    final SettableApiFuture<List<DocumentSnapshot>> futureList = SettableApiFuture.create();
+    final Map<DocumentReference, DocumentSnapshot> documentSnapshotMap = new HashMap<>();
+    for (DocumentReference documentReference : documentReferences) {
+      documentSnapshotMap.put(documentReference, null);
```

Comment:
Add a short code comment explaining why you adding a null

Reply:
i have removed this for loop including setting null now everything will get handled in onCompleted with maintaining order.